[qt] fails to build on macOS 12 (Monterrey)

**Host Environment**
- OS: macOS 12 (Monterrey)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install qt5-base`

**Failure logs** 
```
/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/qt5-base/src/5.15.2-fa8de595e1.clean/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:54:32: error: unknown type name 'CGColorSpaceRef'; did you mean 'QColorSpace'?
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
                               ^~~~~~~~~~~~~~~
                               QColorSpace
/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/qt5-base/src/5.15.2-fa8de595e1.clean/include/QtCore/../../src/corelib/kernel/qmetatype.h:2090:1: note: 'QColorSpace' declared here
QT_FOR_EACH_STATIC_GUI_CLASS(QT_FORWARD_DECLARE_STATIC_TYPES_ITER)
^
/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/qt5-base/src/5.15.2-fa8de595e1.clean/include/QtCore/../../src/corelib/kernel/qmetatype.h:178:24: note: expanded from macro 'QT_FOR_EACH_STATIC_GUI_CLASS'
    F(QColorSpace, 87, QColorSpace) \
                       ^
/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/qt5-base/src/5.15.2-fa8de595e1.clean/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.mm:90:32: error: out-of-line definition of 'setColorSpace' does not match any declaration in 'QIOSurfaceGraphicsBuffer'
void QIOSurfaceGraphicsBuffer::setColorSpace(QCFType<CGColorSpaceRef> colorSpace)
                               ^~~~~~~~~~~~~
/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/qt5-base/src/5.15.2-fa8de595e1.clean/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:54:24: note: type of 1st parameter of member declaration does not match definition ('QCFType<QColorSpace>' vs 'QCFType<CGColorSpaceRef>')
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
                       ^
2 errors generated.
```

--- a/ports/qt5-base/patches/cocoa.patch
+++ b/ports/qt5-base/patches/cocoa.patch
@@ -1,0 +1,12 @@
+diff --git a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+index e070ba97..07c75b04 100644
+--- a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
++++ b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+@@ -40,6 +40,7 @@
+ #ifndef QIOSURFACEGRAPHICSBUFFER_H
+ #define QIOSURFACEGRAPHICSBUFFER_H
+ 
++#include <CoreGraphics/CGColorSpace.h>
+ #include <qpa/qplatformgraphicsbuffer.h>
+ #include <private/qcore_mac_p.h>
+ 

--- a/ports/qt5-base/portfile.cmake
+++ b/ports/qt5-base/portfile.cmake
@@ -81,6 +81,7 @@ qt_download_submodule(  OUT_SOURCE_PATH SOURCE_PATH
                             patches/zstdd.patch                #Fix detection of zstd in debug builds
                             patches/mysql_plugin_include.patch #Fix include path of mysql plugin
                             patches/mysql-configure.patch      #Fix mysql project
+                            patches/cocoa.patch                #Fix missing include on macOS Monterrey, https://code.qt.io/cgit/qt/qtbase.git/commit/src/plugins/platforms/cocoa?id=dece6f5840463ae2ddf927d65eb1b3680e34a547
                             #patches/static_opengl.patch       #Use this patch if you really want to statically link angle on windows (e.g. using -opengl es2 and -static).
                                                                #Be carefull since it requires definining _GDI32_ for all dependent projects due to redefinition errors in the
                                                                #the windows supplied gl.h header and the angle gl.h otherwise.

--- a/ports/qt5-base/vcpkg.json
+++ b/ports/qt5-base/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "qt5-base",
   "version-semver": "5.15.2",
-  "port-version": 13,
+  "port-version": 14,
   "description": "Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
   "dependencies": [

--- a/ports/qtbase/cocoa.patch
+++ b/ports/qtbase/cocoa.patch
@@ -1,0 +1,15 @@
+diff --git a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+index 5d4b6d6a71..cc7193d8b7 100644
+--- a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
++++ b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+@@ -43,6 +43,7 @@
+ #include <qpa/qplatformgraphicsbuffer.h>
+ #include <private/qcore_mac_p.h>
+ 
++#include <CoreGraphics/CGColorSpace.h>
+ #include <IOSurface/IOSurface.h>
+ 
+ QT_BEGIN_NAMESPACE
+-- 
+cgit v1.2.1
+

--- a/ports/qtbase/portfile.cmake
+++ b/ports/qtbase/portfile.cmake
@@ -17,6 +17,7 @@ set(${PORT}_PATCHES
         fix_cmake_build.patch
         harfbuzz.patch
         fix_egl.patch
+        cocoa.patch
         )
 
 if(NOT VCPKG_USE_HEAD_VERSION AND NOT QT_IS_LATEST)

--- a/ports/qtbase/vcpkg.json
+++ b/ports/qtbase/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "qtbase",
   "version-semver": "6.2.1",
+  "port-version": 1,
   "description": "Qt Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5466,7 +5466,7 @@
     },
     "qt5-base": {
       "baseline": "5.15.2",
-      "port-version": 13
+      "port-version": 14
     },
     "qt5-canvas3d": {
       "baseline": "0",
@@ -5634,7 +5634,7 @@
     },
     "qtbase": {
       "baseline": "6.2.1",
-      "port-version": 0
+      "port-version": 1
     },
     "qtcharts": {
       "baseline": "6.2.1",

--- a/versions/q-/qt5-base.json
+++ b/versions/q-/qt5-base.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1e5756de068474651dc8ee50fb14be3d4c236b05",
+      "version-semver": "5.15.2",
+      "port-version": 14
+    },
+    {
       "git-tree": "d0254baec079c6226b73cc7580e7724cad98648c",
       "version-semver": "5.15.2",
       "port-version": 13

--- a/versions/q-/qtbase.json
+++ b/versions/q-/qtbase.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "700e8c2e47b2d044d1623ea5062e15ac6cc9a970",
+      "version-semver": "6.2.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "85405a7d213571f8da152112d436292099c5a4ab",
       "version-semver": "6.2.1",
       "port-version": 0

